soc/xtensa/esp32/CMakeLists.txt

```diff
@@ -5,3 +5,77 @@ zephyr_sources(
   esp32-mp.c
   loader.c
   )
+
+  if(CONFIG_BOOTLOADER_ESP_IDF)
+  include(ExternalProject)
```

Comment:
Nothing wrong on this line, but the commit message in: https://github.com/zephyrproject-rtos/zephyr/pull/37872/commits/2c6aee2cb79d0e42f87a04a673412677037efbe9

doesn't have a single line mentioning why the IDF Bootloader CMake related code is being moved from `boards/xtensa/esp32`  ` soc/xtensa/esp32/`.

Please describe this change in the commit message, as I don't see the relation to the other changes described in that message.

Reply:
@tejlmand, It was moved so `soc` folder as per Marti's request:  
 `Since this support is for the entire esp32 architecture, I similarly think that putting it in a single board directory looks like the wrong architectural design and this should be factored out so it can be reused by all boards in the arch.`  
 I do agree that all this cmake stuff should be placed in `soc` level instead of `board` level. I'll add this into commit message as well.

Reply:
> @tejlmand, It was moved to soc folder as per Marti's request:

yes, I know why because I followed the discussion and I agree it's better to have in the SoC, my point was only to have this reason described in the commit message.

You can't expect people in the future to go back to the PR itself and read 100+ comments to figure out why something was done.